lib/player.js

```diff
@@ -1868,8 +1868,15 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     // Pick the initial streams to play.
     const initialVariant = this.chooseVariant_();
     goog.asserts.assert(initialVariant, 'Must choose an initial variant!');
-    this.switchVariant_(initialVariant, /* fromAdaptation= */ true,
-        /* clearBuffer= */ false, /* safeMargin= */ 0);
+
+    // However, we would skip switch to initial variant
+    // if user already pick variant track (via selectVariantTrack api)
+    const activeVariantTrack = this.getVariantTracks().find((t) => t.active);
+
+    if (!activeVariantTrack) {
```

Comment:
Looking at the rest of the function, it occurs to me that we should probably do the same for text streams below.